### PR TITLE
travis: running mingw build on travis ci

### PR DESCRIPTION
**Pull request by tech4me**
Now do mingw build in travis ci instead of appveyor.
The old bugs with MINGW still exsist.

### CODE REVIEW

File: CMakeModules/MinGWCross.cmake [hunk 1]
```diff
@@ -0,0 +1,54 @@
+SET(MINGW_PREFIX   /usr/x86_64-w64-mingw32/)
```
Comment (lioncash) [inline CMakeModules/MinGWCross.cmake]: The convention with CMake is to use lower-case for command names (... like literally every main CMake file in our codebase)

**Comment by tech4me** [comment]
@lioncash Fixed.

**Comment by Dantttt** [comment]
Lgtm.

File: .travis/linux-mingw/docker.sh [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+mv /bin/uname /bin/uname1 && echo -e '#!/bin/sh\necho MINGW64' >> /bin/uname
+chmod +x /bin/uname
+
+# Dirty hack to trick unicorn makefile into believing we have cmd
+echo '' >> /bin/cmd
+chmod +x /bin/cmd
+
+mkdir build && cd build
+cmake .. -DCMAKE_TOOLCHAIN_FILE="$(pwd)/../CMakeModules/MinGWCross.cmake" -DUSE_CCACHE=ON -DYUZU_USE_BUNDLED_UNICORN=ON -DENABLE_COMPATIBILITY_LIST_DOWNLOAD=ON -DCMAKE_BUILD_TYPE=Release
+make -j4
```
Comment (janisozaur) [inline .travis/linux-mingw/docker.sh]: Why not `ninja`?

**Comment by bunnei** [comment]
@jroweboy can you review please?

**Comment by jroweboy** [comment]
The biggest issues I have with this PR is that it feels like its only a single step forwards, when we already have taken a few more steps in citra. If you compare this with the citra build scripts, you'll see that citra has already moved to using prebuilt docker images that are hosted here https://github.com/citra-emu/build-environments which lets travis cache the docker image with all dependencies pre installed. The has the bonus advantage that ~~if~~ when we move to a self hosted CI solution, that we don't have near as much to migrate hopefully. 

As it *is* an improvement, I'm fine merging it after my comments are addressed. (The code is pretty much the same as what citra had before moving to prebuilt docker images) But I do wish we started pushing for getting those prebuilt images ready for yuzu as well :)

File: CMakeModules/MinGWCross.cmake [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        message("-----------------------------------------")
+        message("* CMAKE_C_COMPILER                     : ${CMAKE_C_COMPILER}")
+        message("* CMAKE_CXX_COMPILER                   : ${CMAKE_CXX_COMPILER}")
+        message("* CMAKE_RC_COMPILER                    : ${CMAKE_RC_COMPILER}")
+        message("* WINDRES                              : ${WINDRES}")
+        message("* ENV{PKG_CONFIG}                      : $ENV{PKG_CONFIG}")
+        message("* STRIP                                : ${STRIP}")
+        message("* USE_CCACHE                           : ${USE_CCACHE}")
+        message("")
+	# So that the debug info only appears once
```
Comment (jroweboy) [inline CMakeModules/MinGWCross.cmake]: wrong indentation

File: .travis/linux/docker.sh [hunk 4]
```diff
@@ -6,7 +6,9 @@ apt-get install --no-install-recommends -y build-essential git libqt5opengl5-dev
 cd /yuzu
 
 mkdir build && cd build
-cmake .. -DYUZU_BUILD_UNICORN=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=/usr/lib/ccache/gcc -DCMAKE_CXX_COMPILER=/usr/lib/ccache/g++ -DENABLE_COMPATIBILITY_LIST_DOWNLOAD=ON -G Ninja
+cmake .. -DYUZU_USE_BUNDLED_UNICORN=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=/usr/lib/ccache/gcc -DCMAKE_CXX_COMPILER=/usr/lib/ccache/g++ -DENABLE_COMPATIBILITY_LIST_DOWNLOAD=ON -G Ninja
```
Comment (jroweboy) [inline .travis/linux/docker.sh]: why is the linux build using `-DYUZU_USE_BUNDLED_UNICORN` all of the sudden?

File: .travis.yml [hunk 5]
```diff
@@ -29,6 +29,19 @@ matrix:
...
       cache: ccache
+    - os: linux
+      env: NAME="MinGW build"
+      sudo: required
+      dist: trusty
+      services: docker
+      addons:
+        apt:
+          packages:
+            - p7zip-full
```
Comment (jroweboy) [inline .travis.yml]: we aren't using 7zip anywhere that i can see

File: .travis/linux-mingw/docker.sh [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+MINGW_PACKAGES="sdl2-mingw-w64 qt5base-mingw-w64 qt5tools-mingw-w64 libsamplerate-mingw-w64 qt5multimedia-mingw-w64"
+apt-get update
+apt-get install -y gpg wget git python3-pip python ccache g++-mingw-w64-x86-64 gcc-mingw-w64-x86-64 mingw-w64-tools cmake
+echo 'deb http://ppa.launchpad.net/tobydox/mingw-w64/ubuntu bionic main ' > /etc/apt/sources.list.d/extras.list
+apt-key adv --keyserver keyserver.ubuntu.com --recv '72931B477E22FEFD47F8DECE02FE5F12ADDE29B2'
+apt-get update
+apt-get install -y ${MINGW_PACKAGES}
+
+# fix a problem in current MinGW headers
+wget -q https://github.com/Alexpux/mingw-w64/raw/d0d7f784833bbb0b2d279310ddc6afb52fe47a46/mingw-w64-headers/crt/errno.h -O /usr/x86_64-w64-mingw32/include/errno.h
```
Comment (jroweboy) [inline .travis/linux-mingw/docker.sh]: not a big fan of downloading it from that url. i think github recommends using the raw.githubusercontent url instead 

https://raw.githubusercontent.com/Alexpux/mingw-w64/d0d7f784833bbb0b2d279310ddc6afb52fe47a46/mingw-w64-headers/crt/errno.h

tbh i like the approach we have for the prebuilt docker images, but this is fine i suppose

File: .travis/linux/docker.sh (hunk 4 above)
Comment (tech4me) [reply .travis/linux/docker.sh]: Rightnow out of the travis ci log on master branch there is:
```
CMake Warning:
  Manually-specified variables were not used by the project:
    YUZU_BUILD_UNICORN
```

I know the option YUZU_BUILD_UNICORN was introduced some time ago but somehow it was removed.

File: .travis.yml (hunk 5 above)
Comment (tech4me) [reply .travis.yml]: 7z is used to compress the artifacts

Comment (jroweboy) [reply .travis.yml]: mmm sorry i forgot that we still produce and upload 7z artifacts for the installer/updater, even though we don't have an installer atm

**Comment by bunnei** [comment]
@jroweboy thanks for the review :) ping to review again now that it looks like comments were addressed.

**Pull request merged**